File: test/python/pytest-to-unittest/parametrize/PARAM-003.expected.py
import pytest
import unittest


class TestCombo(unittest.TestCase):
    # HAMLET-TODO [UNCONVERTIBLE-PARAMETRIZE]: @pytest.mark.parametrize has no direct unittest equivalent
    # Original: @pytest.mark.parametrize("x", [1, 2])
    # Manual action required: Use subTest() or create individual test methods for each parameter set
    @pytest.mark.parametrize("x", [1, 2])
    # HAMLET-TODO [UNCONVERTIBLE-PARAMETRIZE]: @pytest.mark.parametrize has no direct unittest equivalent
    # Original: @pytest.mark.parametrize("y", [3, 4])
    # Manual action required: Use subTest() or create individual test methods for each parameter set
    @pytest.mark.parametrize("y", [3, 4])
    def test_combo(self, x, y):
        self.assertGreater(x + y, 0)

File: test/java/junit5-to-testng/parameterized/PARAM-003.expected.java
import org.testng.Assert;

public class CsvSourceTest {
    // HAMLET-TODO [UNCONVERTIBLE-PARAMETERIZED-TEST]: JUnit 5 @ParameterizedTest requires manual conversion to TestNG @DataProvider
    // Original: @ParameterizedTest
    // Manual action required: Create a @DataProvider method and reference it with @Test(dataProvider = "...")
    @ParameterizedTest
    // HAMLET-TODO [UNCONVERTIBLE-CSV-SOURCE]: JUnit 5 @CsvSource has no direct TestNG equivalent
    // Original: @CsvSource({"1, 1", "2, 4", "3, 9"})
    // Manual action required: Convert CSV data into a @DataProvider method returning Object[][]
    @CsvSource({"1, 1", "2, 4", "3, 9"})
    public void testSquare(int input, int expected) {
        Assert.assertEquals(input * input, expected);
    }
}
